From: Marco Martin <notmart@gmail.com>
Date: Wed, 21 Dec 2016 18:10:04 +0100
Subject: consider an empty panel always "applets loaded"

if a panel doesn't ahve applets, it can't have a "currently
loading" as the status of its applets.
this fixes panels sometimes not showing up when reconnecting screens

CCBUG:373836
---
 src/plasma/containment.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/plasma/containment.cpp b/src/plasma/containment.cpp
index 133220a6a..b9c8c8b56 100644
--- a/src/plasma/containment.cpp
+++ b/src/plasma/containment.cpp
@@ -297,6 +297,10 @@ void Containment::restoreContents(KConfigGroup &group)
         d->createApplet(plugin, QVariantList(), appId);
     }
 
+    //if there are no applets, none of them is "loading"
+    if (Containment::applets().isEmpty()) {
+        d->appletsUiReady = true;
+    }
     foreach (Applet *applet, Containment::applets()) {
         if (!applet->pluginMetaData().isValid()) {
             applet->updateConstraints(Plasma::Types::UiReadyConstraint);
